src/test/txvalidationcache_tests.cpp

```diff
@@ -165,12 +166,12 @@ BOOST_FIXTURE_TEST_CASE(uncache_coins, TestChain100Setup)
     // Add an orphan to the orphan cache.  The valid inputs should be present in the coins cache.
     spends.resize(3);
     spends[2].vin.resize(3);
-    spends[2].vin[0].prevout.hash = GetRandHash();
-    spends[2].vin[0].prevout.n = 0;
+    spends[2].vin[2].prevout.hash = GetRandHash();
+    spends[2].vin[2].prevout.n = 0;
     spends[2].vin[1].prevout.hash = GetRandHash();
     spends[2].vin[1].prevout.n = 0;
-    spends[2].vin[2].prevout.hash = coinbaseTxns[2].GetHash();
-    spends[2].vin[2].prevout.n = 0;
+    spends[2].vin[0].prevout.hash = coinbaseTxns[2].GetHash();
+    spends[2].vin[0].prevout.n = 0;
     spends[2].vout.resize(1);
```

Comment:
don't get why we need to change `vin` order here

Reply:
I wish that I could write a comment in the code to explain a change but then it automatically ages out or something.

In the mempool validation code, it used to check every input even if it found a problem in the first one.  I changed that to abort at the first missing input.  Because of this optimization, input 2 won't be loaded into the cache if input 0 was bad.

This test is checking proper cache loading for some inputs.  I need to put the inputs that the test is checking first, before bad ones.